### ISSUE TITLE
Method decodeToken has an error in charset

### ISSUE THREAD

**Issue by mwiesmueller** [auth0/angular2-jwt#174]
The method decodeToken decode the charset of the payload not correct. The key currencysymbol must be equal to '€'. (See snippet)

In Version 0.1.17 was everything ok with this method. After update to 0.1.22 or 0.1.23 we've got this problem.

Has everything changed in this method? Where is my error? :-)

Here a snippet of the payload with the error:
`{
"employeeid":2,
"subsidiaries":[
{
"currency":"EUR",
"currencysymbol":"â¬"
}
],
"exp":1474477088
}`

**Comment by escardin** [comment]
can you provide an encoded token we can use to test with? If I had to guess though, we're decoding with either utf-8 or some sort of default, and the string you're sending is cp-1252 or similar.

**Comment by mwiesmueller** [comment]
No Problem.. Here is a token from my system. See the key currencysymbol in subsidiaries array:

Thank you very much..

`eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbXBsb3llZWlkIjoyLCJ1c2VybmFtZSI6InBvaGxAd2VyYmFzaW5ub3RlYy5jb20iLCJmaXJzdG5hbWUiOiJNaWNoYWVsIiwibGFzdG5hbWUiOiJQb2hsIiwiY29tcGFueW5yIjoxLCJkZWJpdG9yaWQiOjk5MTM2OSwic2FsdXRhdGlvbiI6IkhlcnIiLCJzYWx1dGF0b3J5YWRkcmVzcyI6IlNlaHIgZ2VlaHJ0ZXIgSGVyciBQb2hsLCIsInN1YnNpZGlhcmllcyI6W3sibmFtZSI6IkhhdXB0ZmlsaWFsZSIsImlkIjoxLCJjdXJyZW5jeSI6IkVVUiIsImN1cnJlbmN5c3ltYm9sIjoi4oKsIiwicGhvbmUiOiIwNzAzMS0yMjIyMC0wIiwiZW1haWwiOiJtYXJ0aW4ud2llc211ZWxsZXJAd2VyYmFzLWFnLmNvbSIsImFkZHJlc3MiOnsibGluZTEiOiJBdXRvIFdlYmVyIiwibGluZTIiOiJJbmguIFJhbGYgV2ViZXIiLCJsaW5lMyI6IiIsInN0cmVldCI6Ik1heC1FeXRoLVN0ci4gMjQiLCJjb3VudHJ5Y29kZSI6IkRFIiwiemlwY29kZSI6IjcxMDg4IiwiY2l0eSI6IkhvbHpnZXJsaW5nZW4ifX0seyJuYW1lIjoiTmViZW5maWxpYWxlIiwiaWQiOjIsImN1cnJlbmN5IjoiRVVSIiwiY3VycmVuY3lzeW1ib2wiOiLigqwiLCJwaG9uZSI6IiIsImVtYWlsIjoiIiwiYWRkcmVzcyI6eyJsaW5lMSI6IldlcmJhcyBOZWJlbmZpbGlhbGUiLCJsaW5lMiI6IiIsImxpbmUzIjoiIiwic3RyZWV0IjoiTWF4LUV5dGgtU3RyLiA0MiIsImNvdW50cnljb2RlIjoiREUiLCJ6aXBjb2RlIjoiNzEwODgiLCJjaXR5IjoiSG9semdlcmxpbmdlbiJ9fV0sImV4cCI6MTQ3NDQ4ODAzOSwicmVhZGFibGVleHBpcmVkIjoiMjEuMDkuMjAxNiAyMDowMDozOSIsIm0ybXN5c3RlbSI6IldFUkJBU19taW5pIn0.1dJ8Ng0rgTgtqbQ_J6wKFkxMJRCHKFWWXVFE_FUjciw`

**Comment by mwiesmueller** [comment]
Additional Information:

with version 0.1.17 everything is ok. I've just checked both version on the same backend. With the older version the token will correct decoded from the method. With the newer version happened the error.

**Comment by escardin** [comment]
Okay, so I have a pr ready for this. https://github.com/auth0/angular2-jwt/pull/176

It boils down to after .17 (commit 6c4223b593c242df35c1ce3163e06508ffd0fe2a) someone changed from the deprecated js builtin `escape()` to `encodeURI()` which doesn't actually do the same thing at all (neither does encodeURIComponent()).

I changed the whole decoding bit to use a lib that does it properly, but I am not really happy with the ~3kb it adds when minified.

I found that your entire token was somehow making jasmine unhappy with my copy pasted decoded version, so I just boiled the new test down to the symbol.

I found a workaround on MDN, but it didn't actually work with a pre-encoded token https://developer.mozilla.org/en/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem

@mwiesmueller I'm not sure if it's better to be unfair to you and keep the total library payload smaller while forcing you to decode your own tokens, or to bloat the lib somewhat. JS is hard. :S

It's probably safe to to the escape trick indefinitely, even if it's deprecated. I expect JS to end up like java with all the bad bits that were actually used never being removed.
